File: backend/read_assets.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import openpyxl
import json
from datetime import datetime
import sys

# Read Excel file
excel_file = r'c:\Apps\assethub-V2\AssetIT41.xlsx'

try:
    wb = openpyxl.load_workbook(excel_file)
    ws = wb.active
    
    print("=" * 80)
    print(f"📊 Asset File: {excel_file}")
    print("=" * 80)
    
    # Get headers
    headers = []
    for cell in ws[1]:
        headers.append(cell.value)
    
    print(f"\n📋 Headers (Column {len(headers)}):")
    for i, h in enumerate(headers, 1):
        print(f"  {i:2d}. {h}")
    
    # Get data rows
    assets = []
    print(f"\n📦 Data Preview (First 10 rows):")
    print("-" * 80)
    
    for row_idx, row in enumerate(ws.iter_rows(min_row=2, values_only=False), start=2):
        if row_idx > 11:  # Show only first 10 data rows
            break
        
        row_data = {}
        for col_idx, cell in enumerate(row):
            header = headers[col_idx] if col_idx < len(headers) else f"Col_{col_idx}"
            value = cell.value
            row_data[header] = value
        
        assets.append(row_data)
        
        print(f"\nRow {row_idx}:")
        for key, val in row_data.items():
            if val is not None:
                print(f"  {key:30s}: {val}")
    
    # Count total rows
    total_rows = ws.max_row - 1
    print(f"\n{'=' * 80}")
    print(f"✓ Total records: {total_rows}")
    print(f"{'=' * 80}")
    
    # Save to JSON for TypeScript processing
    json_output = {
        'headers': headers,
        'total_rows': total_rows,
        'sample_rows': assets
    }
    
    with open(r'c:\Apps\assethub-V2\backend\asset_data.json', 'w', encoding='utf-8') as f:
        json.dump(json_output, f, indent=2, ensure_ascii=False, default=str)
    
    print(f"\n✓ JSON saved to: c:\\Apps\\assethub-V2\\backend\\asset_data.json")
    
except Exception as e:
    print(f"❌ Error: {e}", file=sys.stderr)
    sys.exit(1)

File: backend/src/read-assets.ts
import * as XLSX from 'xlsx';
import * as fs from 'fs';

const excelFile = 'c:\\Apps\\assethub-V2\\AssetIT41.xlsx';

try {
  console.log('\n========================================');
  console.log('📊 Reading Asset File...');
  console.log('========================================\n');

  // Read Excel file
  const workbook = XLSX.readFile(excelFile);
  const worksheet = workbook.Sheets[workbook.SheetNames[0]];
  const data = XLSX.utils.sheet_to_json(worksheet, { defval: '' }) as any[];

  console.log(`📋 Sheet Name: ${workbook.SheetNames[0]}`);
  console.log(`📦 Total Records: ${data.length}`);

  // Display headers
  if (data.length > 0) {
    const headers = Object.keys(data[0]);
    console.log(`\n🔑 Columns (${headers.length}):`);
    headers.forEach((h, i) => {
      console.log(`  ${(i + 1).toString().padStart(2, ' ')}. ${h}`);
    });

    // Display sample data
    console.log('\n📋 Sample Data (First 5 rows):');
    console.log('----------------------------------------');
    data.slice(0, 5).forEach((row, idx) => {
      console.log(`\nRow ${idx + 1}:`);
      Object.entries(row).forEach(([key, value]) => {
        if (value) {
          console.log(`  ${key.padEnd(30, ' ')}: ${value}`);
        }
      });
    });
  }

  // Save data for processing
  fs.writeFileSync(
    'c:\\Apps\\assethub-V2\\backend\\asset_data.json',
    JSON.stringify({ headers: Object.keys(data[0] || {}), data, total: data.length }, null, 2)
  );

  console.log('\n✓ Data saved to: asset_data.json');
  console.log('========================================\n');

} catch (err) {
  console.error('❌ Error:', err);
  process.exit(1);
}
